### PR TITLE
feat(PT-2813): x2y2 order validation

### PR DESCRIPTION
**Pull request by dmitry-saraykin**

### CODE REVIEW

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderX2Y2StatusTaskHandler.kt [hunk 1]
```diff
@@ -37,33 +34,11 @@ class OrderX2Y2StatusTaskHandler(
...
-            caller = order.maker,
-            orderId = data.orderId,
-            currency = order.currency.token,
-            price = price.value,
-            tokenId = tokenId?.value ?: throw IllegalStateException("Can't get tokenId for order: ${order.hash}"),
-        )
+        val isActive = x2y2Service.isActiveOrder(order)
         if (isActive.not() && properties.fixX2Y2) {
             logger.info(
-                "Detected not active x2y2 order ${order.hash}, collection=${order.token}, tokeId=${tokenId}"
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderX2Y2StatusTaskHandler.kt:66]: Better to log in itemId format (collection:tokenId.bigInteger) as for all items

**Review by sanufriev — APPROVED** [review]

File: order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt [hunk 2]
```diff
@@ -125,30 +129,67 @@ class OrderService(
...
         if (!featureFlags.enableOrderValidation) {
             return order
         }
 
-        if (order.status !== OrderStatus.ACTIVE) {
-            logger.warn("Order validation error: hash=$hash, status=${order.status}")
-            orderUpdateService.update(hash, eventTimeMarks)
-            throw ValidationApiException("order is not active")
+        checkStatus(order, hash)
+        checkApprovals(order, hash)
```
Comment (yakischik) [inline order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt:140]: Maybe we can do it as List<OrderChecker> injection? It's easier add new aggragation checkers and tests are also more targeted in such case

File: order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt [hunk 3]
```diff
@@ -125,30 +129,67 @@ class OrderService(
...
         }
-        if (order.platform == Platform.OPEN_SEA && order.data.version == OrderDataVersion.BASIC_SEAPORT_DATA_V1) {
-            orderSignatureResolver.resolveSeaportSignature(hash)
+    }
+
+    private suspend fun checkStatus(order: Order, hash: Word) {
+        if (order.status !== OrderStatus.ACTIVE) {
+            logger.warn("Order validation error: hash=$hash, status=${order.status}")
+            orderUpdateService.update(hash, orderOffchainEventMarks())
+            throw ValidationApiException("order is not active")
```
Comment (yakischik) [inline order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt:191]: Lets add hash and platform in response

File: order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt [hunk 4]
```diff
@@ -125,30 +129,67 @@ class OrderService(
...
+        checkX2Y2(order)
+        return order
+    }
+
+    private suspend fun checkX2Y2(order: Order) {
+        if (order.platform == Platform.X2Y2) {
+            val active = try {
+                x2y2Service.isActiveOrder(order)
+            } catch (e: Exception) {
+                logger.error("Error during getting x2y2 order status: $e", e)
```
Comment (yakischik) [inline order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt:151]: Lets log order hash and itemId here

File: order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt [hunk 5]
```diff
@@ -125,30 +129,67 @@ class OrderService(
...
+                true
+            }
+            if (!active) {
+                orderCancelService.cancelOrder(id = order.hash, eventTimeMarksDto = orderOffchainEventMarks())
+                throw OrderDataException("order is not active")
```
Comment (yakischik) [inline order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt:156]: Lets add order hash and platform to the response

File: order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt (hunk 2 above)
Comment (dmitry-saraykin) [reply order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt:140]: done in the next pr https://github.com/rarible/ethereum-indexer/pull/660/files#diff-28673cfc00a2e67330f941403e309c6081bf95ac76cbca2945344e3a26de3ec6R6

File: order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt (hunk 3 above)
Comment (dmitry-saraykin) [reply order/api/src/main/kotlin/com/rarible/protocol/order/api/service/order/OrderService.kt:191]: https://github.com/rarible/ethereum-indexer/pull/660/files#diff-bd479947f8c06b334749d489c7b35653d64be24f3c9f058f5908f0ec9aa8d7f2R20

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderX2Y2StatusTaskHandler.kt (hunk 1 above)
Comment (dmitry-saraykin) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderX2Y2StatusTaskHandler.kt:66]: https://github.com/rarible/ethereum-indexer/pull/660/files#diff-1c71d59c66512af8c1c78b7abe2ae28636630baa276de6cde4c7775aea49db44R41

**Pull request merged**